### ISSUE TITLE
Provide tenant DisplayName and DefaultDomain on az account commands or provide a command to fetch tenant information

### ISSUE THREAD

**Issue by jikuja** [Azure/azure-cli#19276]
**Is your feature request related to a problem? Please describe.**
Selecting correct subscription from `az account list` is hard if identical subscription names are used on multiple tenants because: 
1. The command does not output tenants' DisplayName and DefaultDomain
2. AZ CLI does not provide a command to list tenants

**Describe the solution you'd like**
`az account list` and `az account show` should display tenants' DisplayName and DefaultDomain.

**Alternative solution**
Add a command that gives the content of the following REST call: `az rest --method GET --uri https://management.azure.com/tenants?api-version=2020-01-01 --output table --query 'value[]'`. Compare with Get-AzTenant PS Cmdlet

The best-case scenario would be implementing both.

**Comment by yonzhan** [comment]
@jiasli for awareness

**Comment by jiasli** [comment]
Indeed, Azure CLI calls [Tenants - List](https://docs.microsoft.com/en-us/rest/api/resources/tenants/list) to get all tenants the user has access to.

The REST response does contain `displayName`, `defaultDomain`, along with other information.

```
{
  "value": [
    {
      "id": "/tenants/a70a1586-9c4a-4373-b907-1d310660dbd1",
      "tenantId": "a70a1586-9c4a-4373-b907-1d310660dbd1",
      "countryCode": "US",
      "displayName": "Test_Test_aad50",
      "domains": [
        "aad50.ccsctp.net"
      ],
      "tenantCategory": "ManagedBy",
      "defaultDomain": "aad50.ccsctp.net",
      "tenantType": "AAD"
    },
    ...
```

We will consider this as a feature request.

**Comment by jiasli** [comment]
On a second thought, [Tenants - List](https://docs.microsoft.com/en-us/rest/api/resources/tenants/list) is only called by

https://github.com/Azure/azure-cli/blob/397db42b035255ec59812dd5f71586a6693c1c2b/src/azure-cli-core/azure/cli/core/_profile.py#L720

On the other hands, if `--tenant` is provided to `az login`, Azure CLI directly gets an access token for the provided tenant and calls

https://github.com/Azure/azure-cli/blob/397db42b035255ec59812dd5f71586a6693c1c2b/src/azure-cli-core/azure/cli/core/_profile.py#L794

thus having no chance to get the tenant information. If CLI wants to get tenant information, an additional call to [Tenants - List](https://docs.microsoft.com/en-us/rest/api/resources/tenants/list) will be necessary.

Also, there is no **Tenants - Get** REST API.

![image](https://user-images.githubusercontent.com/4003950/179148019-4744f98a-e853-45c8-9fec-3b11e7b2061f.png)
